src/solvers/dg_simplices/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+"""
+  function hadamard_sum_ATr!(du, ATr, volume_flux, u, skip_index=(i,j)->false)
+
+Computes the flux difference ∑_j Aij * f(u_i, u_j) and accumulates the result into `du`. 
+- `du`, `u` are vectors
+- `ATr` is the transpose of the flux differencing matrix `A`. The transpose is used for 
+faster traversal since matrices are column major in Julia. 
+"""
+@inline @muladd function hadamard_sum_ATr!(du, ATr, volume_flux::VF, u, skip_index=(i,j)->false) where {VF}
+  rows,cols = axes(ATr)
+  for i in cols
+    ui = u[i]
+    val_i = du[i]
+    for j in rows
+      if !skip_index(i,j)                
+          val_i += ATr[j,i] * volume_flux(ui, u[j])
+      end
+    end
+    du[i] = val_i 
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{2, Tri}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  @unpack rxJ, sxJ, ryJ, syJ = mesh.md
+  QrskewTr, QsskewTr = Qrst_skew_Tr
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem] * QrskewTr + sxJ[1,elem] * QsskewTr))
+  else
+    return LazyArray(@~ @. 2 * (ryJ[1,elem] * QrskewTr + syJ[1,elem] * QsskewTr))
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{3, Tet}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  QrskewTr, QsskewTr, QtskewTr = Qrst_skew_Tr
+  @unpack rxJ, sxJ, txJ, ryJ, syJ, tyJ, rzJ, szJ, tzJ = mesh.md
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem]*QrskewTr + sxJ[1,elem]*QsskewTr + txJ[1,elem]*QtskewTr))
+  elseif orientation == 2
+    return LazyArray(@~ @. 2 * (ryJ[1,elem]*QrskewTr + syJ[1,elem]*QsskewTr + tyJ[1,elem]*QtskewTr))
+  elseif orientation == 3
+    return LazyArray(@~ @. 2 * (rzJ[1,elem]*QrskewTr + szJ[1,elem]*QsskewTr + tzJ[1,elem]*QtskewTr))
+  end
+end
+
+function calc_volume_integral!(du, u::StructArray, volume_integral::VolumeIntegralFluxDifferencing,
+                 mesh::VertexMappedMesh, equations, dg::DG, cache) where {DG <: SBPDGFluxDiff}
+
+  rd = dg.basis  
+  @unpack local_values_threaded = cache
+
+  volume_flux_oriented(i) = let i=i, equations=equations 
+    @inline (u_ll, u_rr)->volume_integral.volume_flux(u_ll, u_rr, i, equations)
+  end
```

Comment:
`let` blocks inside your closure shouldn't do anything, do they?

Reply:
I'm not sure? Because of https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-captured, I usually use `let` blocks when using closures.

Reply:
We discussed this in one of your earlier drafts: https://github.com/trixi-framework/Trixi.jl/pull/557#issuecomment-830581577 and https://github.com/trixi-framework/Trixi.jl/commit/e78d468b5ed560adac4a944859ce64e843508a6d. Using the `let` block inside the closure can still be problematic and you should use it outside of the closure. However, I would prefer checking whether the performance bug still applies to your case before doing this optimization. If it is indeed necessary, could you please add a brief comment explaining that? We have quite a lot of students working with Trixi and we can't expect them to know all performance bugs of Julia

Reply:
Gotcha. I'll double check the performance and remove the `let` blocks if this doesn't make a difference in practice.